### PR TITLE
Fixes to statistical recipes

### PR DESCRIPTION
**Pull request by piever**
# Description

Fixes `qqnorm`, `qqplot`, which were broken as they were converted to an odd `Scatter` plot type and that broke with the new internals. They now have a more sensible "standard" implementation as a new plot type.

It also fixed plotting `StatsBase.Histogram` objects. They accidentally ended up having a gap between bars after ,#1223.

## Type of change

Delete options that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Added or changed relevant sections in the documentation

I have not added tests as it seems tests exist already (see https://github.com/JuliaPlots/Makie.jl/blob/master/test/statistical_tests.jl) but somehow they stopped being run. It is probably best to update those in a separate PR (it seems all of them need fixing, due to changes in the internals).

### CODE REVIEW

**Review by SimonDanisch — APPROVED** [review]
Not really the right person to review this ;) If you have opinions on this @jkrumbiegel, please comment as well..Otherwise, I'd just leave this up to @piever  to merge!

File: src/stats/distributions.jl
Comment (piever) [reply src/stats/distributions.jl:41]: Yes, I'm renaming `:quantile` to `:robust` (that's how Base R calls it, see [here](https://www.rdocumentation.org/packages/EnvStats/versions/2.3.1/topics/qqPlot)) under `qq.line.type`). They say `"least squares"` instead of `:fit`, but I think that's a bit too technical.

File: src/stats/distributions.jl
Comment (piever) [reply src/stats/distributions.jl]: Actually, it looks like Base R (see [here](https://www.rdocumentation.org/packages/EnvStats/versions/2.3.1/topics/qqPlot)) and ggplot2 (see [here](https://ggplot2.tidyverse.org/reference/geom_qq.html)) have different defaults, so I'm just omitting that part.

**Comment by piever** [comment]
Thanks for the review! I've addressed the feedback and added a NEWS entry on the renaming of the options for `qqline` .

**Comment by piever** [comment]
Actually, I do have one doubt regarding the default value for `qqline`. I realize that no single choice is ideal, because `qqline=:robust` makes a lot of sense if you are doing the standard `qqnorm` (comparing with a standard Gaussian, or more generally a distribution with default parameters), whereas `qqline=:identity` is IMO a more reasonable default if you are comparing two empirical distributions (from vectors of samples) or data versus a fitted distribution.

OTOH, I don' really like magically switching from one default to the other depending on the argument types. Should we just default to `qqline=:none`, or is `qqline=:robust` a good enough default?

**Comment by palday** [comment]
@piever I like the idea of omitting the line by default and forcing the user to take an explicit stand on which type they want.

File: src/stats/distributions.jl
Comment (palday) [inline src/stats/distributions.jl:78]:
```suggestion
Shorthand for `qqplot(Normal(0,1), y)`. See `qqplot` for more details.
```

To make it very clear to the users (and later devs updating the code :wink:) that it's the standard normal. cf #1277

Comment (piever) [reply src/stats/distributions.jl]: Good point, thank you!

**Comment by piever** [comment]
Planning to merge this in a few days if there are no additional concerns. To summarize the main API changes
- old `qqline=:quantile` or `qqline=:R` are now `qqline=:fitrobust`
- `:none` is the default (as in general it is unclear exactly which line should be drawn)
- a value of `qqline` not among the valid options (`:none, :fit, :fitrobust, :identity`) errors rather than being ignored silently (which allows us to add options in a non-breaking way in the future)

**Pull request merged**
